src/AMOEBA/pair_amoeba.cpp

```diff
@@ -827,28 +827,35 @@ void PairAmoeba::init_style()
 
   // check if all custom atom arrays were set via fix property/atom
 
-  int flag,cols;
-
-  index_amtype = atom->find_custom("amtype",flag,cols);
-  if (index_amtype < 0 || flag || cols)
-    error->all(FLERR,"Pair {} amtype is not defined", mystyle);
-  index_amgroup = atom->find_custom("amgroup",flag,cols);
-  if (index_amgroup < 0 || flag || cols)
-    error->all(FLERR,"Pair {} amgroup is not defined", mystyle);
-
-  index_redID = atom->find_custom("redID",flag,cols);
-  if (index_redID < 0 || !flag || cols)
-    error->all(FLERR,"Pair {} redID is not defined", mystyle);
-  index_xyzaxis = atom->find_custom("xyzaxis",flag,cols);
-  if (index_xyzaxis < 0 || !flag || cols == 0)
-    error->all(FLERR,"Pair {} xyzaxis is not defined", mystyle);
-
-  index_polaxe = atom->find_custom("polaxe",flag,cols);
-  if (index_polaxe < 0 || flag || cols)
-    error->all(FLERR,"Pair {} polaxe is not defined", mystyle);
-  index_pval = atom->find_custom("pval",flag,cols);
-  if (index_pval < 0 || !flag || cols)
-    error->all(FLERR,"Pair {} pval is not defined", mystyle);
+  char const * names[6] = {"amtype", "amgroup", "redID",
+    "xyzaxis", "polaxe", "pval"};
+  int const flag_check[6] = {0, 0, 1, 1, 0, 1}; // correct type (0 int, 1 dbl)
+  int const cols_check[6] = {0, 0, 0, 3, 0, 0}; // xyzaxis 3 cols, all others 0
+  int const ghost_check[6] = {1, 0, 0, 0, 0, 0}; // which types need ghost
```

Comment:
In examples/amoeba/in.ubiquitin, 4 of the 6 property/atom vecs have ghost set.  Are you thinking  only one of them needs it?  I just spot-checked amgroup and I think it needs the ghost setting.

Reply:
I've struggled to check this myself -- I can only get the water dimer and hexamer to run on my laptop, not the water box or ubiquitin. I've turned on the ghost check for `amgroup` and `pval` on the heuristic that, like `amtype`, there are settings in `pack_forward_comm` and `unpack_forward_comm` for those property/atom vecs.

Reply:
> Looks like a bug was introduced some time earlier. This line in pair_amoeba.cpp
> if (first_flag_compute && (comm->me == 0)) print_settings();
> should be
> if (first_flag_compute) print_settings();

This kind of construct has been in place from the very beginning. The inconsistent MPI behavior is *caused* by *calling* `final_accuracy_mpole()` which was added in commit 288629bd139 recently.

> @akohlmey - please verify that change is compatible with the intent of the refactoring of print_settings() ?

It needs additional changes to `print_settings()`. I will add them.

Please also note that compilation currently fails to complete due to missing changes to the KOKKOS package code.